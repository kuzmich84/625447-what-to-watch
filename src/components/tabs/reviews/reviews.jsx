import React from "react";
import Review from "../../review/review";
import PropTypes from "prop-types";
import {getIsLoadingReviews, getReviews} from "../../../store/selectors";
import {connect} from "react-redux";

const Reviews = ({reviews, isLoadingReviews}) => {
  const sortReviews = reviews.sort((a, b) => b.rating - a.rating);

  if (isLoadingReviews) {
    return <p>Loading</p>;
  } else {
    return (
      <div className="movie-card__reviews movie-card__row">
        <div className="movie-card__reviews-col">
          {sortReviews.filter((review) => review.id % 2 !== 0).map((review) => <Review
            key={review.id}
            comment={review.comment}
            date={review.date}
            rating={review.rating}
            userName={review.user.name}
          />)}
        </div>
        <div className="movie-card__reviews-col">
          {sortReviews.filter((review) => review.id % 2 === 0).map((review) => <Review
            key={review.id}
            comment={review.comment}
            date={review.date}
            rating={review.rating}
            userName={review.user.name}
          />)}
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  reviews: getReviews(state),
  isLoadingReviews: getIsLoadingReviews(state),
});

export {Reviews};
export default connect(mapStateToProps)(Reviews);

Reviews.propTypes = {
  reviews: PropTypes.arrayOf(PropTypes.object.isRequired)
};
