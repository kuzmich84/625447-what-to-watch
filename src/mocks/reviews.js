export default [
  {
    "id": 1,
    "user": {
      "id": 1,
      "name": `Oliver.conner`
    },
    "rating": 8.9,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2019-05-08T14:13:56.569Z`
  },
  {
    "id": 2,
    "user": {
      "id": 2,
      "name": `Katy`
    },
    "rating": 2,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2020-05-08T14:13:56.569Z`
  },
  {
    "id": 3,
    "user": {
      "id": 3,
      "name": `Boris`
    },
    "rating": 5.5,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2020-08-08T14:13:56.569Z`
  },

];
