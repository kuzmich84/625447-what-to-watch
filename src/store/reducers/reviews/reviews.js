import {ActionType} from "../../action";
import {extend} from "../../../utils";

const initialState = {
  reviews: [],
  isLoadingReview: true,
  isSendReview: false,
  error: null,
};

const reviews = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_REVIEWS:
      return extend(state, {
        reviews: action.payload
      });
    case ActionType.SET_IS_LOADING_REVIEW:
      return extend(state, {
        isLoadingReview: action.payload
      });
    case ActionType.SET_IS_SEND_REVIEW:
      return extend(state, {
        isSendReview: action.payload
      });
    case ActionType.SET_ERROR_REVIEWS:
      return extend(state, {
        error: action.payload,
      });
    default:
      return state;
  }
};

export default reviews;

