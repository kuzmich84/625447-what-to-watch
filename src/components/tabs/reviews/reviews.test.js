import React from "react";
import renderer from "react-test-renderer";
import {Reviews} from "./reviews";
import mockReviews from "../../../mocks/reviews";


test(`Should Reviews component render correctly`, () => {
  const tree = renderer
    .create(<Reviews
      reviews={mockReviews}
      isLoadingReviews={false}
    />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
