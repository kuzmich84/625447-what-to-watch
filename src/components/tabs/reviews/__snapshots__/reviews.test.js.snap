// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Reviews component render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Oliver.conner
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08T14:13:56.569Z"
          >
            May 8, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Boris
          </cite>
          <time
            className="review__date"
            dateTime="2020-08-08T14:13:56.569Z"
          >
            August 8, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.5
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Katy
          </cite>
          <time
            className="review__date"
            dateTime="2020-05-08T14:13:56.569Z"
          >
            May 8, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        2
      </div>
    </div>
  </div>
</div>
`;
